test/acceptance/integration/tools/general_utils.py

```diff
@@ -153,3 +154,160 @@ def drop_all_test_databases(ip, port):
+                empty_attrs = ['ssh_port', 'user', 'password', 'bin_path']
+                for attr in empty_attrs:
+                    if checking[attr] == "":
+                        raise_property_bad_configuration('deploy_data')
+                try:
+                    int(checking['ssh_port'])
+                except ValueError:
+                    print 'The port has to be a string with numbers, but is {port}'.format(port=checking['ssh_port'])
+                    raise_property_bad_configuration('deploy_data')
+    except KeyError as e:
+        print "There is a property bad configured/set: properties: {config}".format(config=world.config)
+        raise e
+
+
+def set_ssh_config(localhost):
+    """
+    Config fabric parms
+    :return:
+    """
+    output['stdout'] = False
+    output['running'] = False
+    output['warnings'] = False
+    try:
+        config = world.config['deploy_data']
+        if not localhost:
+            # Set ssh connection info
+            env.host_string = '{ip}:{ssh_port}'.format(ip=config['host'], ssh_port=config['ssh_port'])
+            env.user = config['user']
+            env.password = config['password']
+            env.sudo_password = config['password']
+            env.warn_only = True
+    except KeyError as e:
+        print 'set_ssh_config: In the properties file there is not definition about the deploy information. Config: {config}\n'.format(
+            config=world.config)
+        raise e
+
+def get_cb_pid():
+    """
+    Get the cb pid, if its running
+    :return:
+    """
+    config = world.config['deploy_data']
+    if not config['host'] == 'localhost' or not config['host'] == '127.0.0.1':
+        localhost = False
+    else:
+        localhost = True
+    set_ssh_config(localhost)
+    return run("ps -ef | grep contextBroker | grep -v grep | awk '{print $2}'")
+
+def start_cb(parms):
+    """
+    Start the Context Broker in the machine set in the properties file
+    :param parms:
+    :return:
+    """
+    config = world.config['deploy_data']
+    if not config['host'] == 'localhost' or not config['host'] == '127.0.0.1':
+        localhost = False
+    else:
+        localhost = True
+    set_ssh_config(localhost)
+    if world.cb_pid != '':
+        stop_cb()
+    if localhost:
+        local('{bin_path} {parms} >> /tmp/cb_acceptance_test.log'.format(bin_path=config['bin_path'], parms=parms))
+    else:
+        run('{bin_path} {parms} >> /tmp/cb_acceptance_test.lo'.format(bin_path=config['bin_path'], parms=parms))
+    world.cb_pid = get_cb_pid()
+
+
+
+def stop_cb():
+    """
+    Stop a cb instance with the pid in world.cb_pid
+    :return:
+    """
+    config = world.config['deploy_data']
+    if not config['host'] == 'localhost' or not config['host'] == '127.0.0.1':
+        localhost = False
+    else:
+        localhost = True
+    set_ssh_config(localhost)
+    # Check if there is CB running
+    if localhost:
```

Comment:
Regarding stopping strategy... CB stores its pid in a file (check `-pidpath` CLI argument). Maybe you should start checking that before trying the ps/grep way.